apis/v1alpha1/httproute_types.go

```diff
@@ -56,7 +56,8 @@ type HTTPRouteHost struct {
 	// Support: Core
 	//
 	// +optional
-	Hostnames []string `json:"hostnames,omitempty"`
+	// +kubebuilder:validation:MaxItems=16
```

Comment:
same question I had about 1:N vs N:1 Gateways here

Reply:
Basically, if we need more than 16 can we easily just split off another httproute and get the same behavior

Reply:
Yeah, absolutely. This is a rather arbitrary limit and I'm happy to increase it if there are use cases that will run into it. Just trying to enforce reasonable upper limits wherever we can.

Reply:
@robscott should a maximum # of `Rules` and/or `ExtensionRef` also be established at this time?

Reply:
Updated to add a max # of rules as well (16).